packages/common/docs/Palette.md

```diff
@@ -0,0 +1,357 @@
+    "kind": 32,
+    "kindString": "Variable",
+    "flags": {
+        "isExported": true,
+        "isConst": true
+    },
+    "sources": [
+        {
+            "fileName": "Palette.ts",
+            "line": 319,
+            "character": 20
+        }
+    ],
+    "type": {
+        "type": "reference",
+        "name": "fetchRainbowItem",
+        "id": 19890
+    },
+    "defaultValue": " fetchRainbowItem",
+    "folder": "packages/common",
+    "signatures": [
+        {
+            "id": 19891,
+            "name": "fetchRainbowItem",
+            "kind": 4096,
+            "kindString": "Call signature",
+            "flags": {},
+            "type": {
+                "type": "array",
+                "elementType": {
+                    "type": "intrinsic",
+                    "name": "string"
+                }
+            }
+        },
+        {
+            "id": 19892,
+            "name": "fetchRainbowItem",
+            "kind": 4096,
+            "kindString": "Call signature",
+            "flags": {},
+            "parameters": [
+                {
+                    "id": 19893,
+                    "name": "id",
+                    "kind": 32768,
+                    "kindString": "Parameter",
+                    "flags": {},
+                    "type": {
+                        "type": "intrinsic",
+                        "name": "string"
+                    }
+                },
+                {
+                    "id": 19894,
+                    "name": "colors",
+                    "kind": 32768,
+                    "kindString": "Parameter",
+                    "flags": {},
+                    "type": {
+                        "type": "array",
+                        "elementType": {
+                            "type": "intrinsic",
+                            "name": "string"
+                        }
+                    }
+                },
+                {
+                    "id": 19895,
+                    "name": "steps",
+                    "kind": 32768,
+                    "kindString": "Parameter",
+                    "flags": {
+                        "isOptional": true
+                    },
+                    "type": {
+                        "type": "intrinsic",
+                        "name": "number"
+                    }
+                }
+            ],
+            "type": {
+                "type": "reference",
+                "name": "rainbowPalleteFunc",
+                "id": 19884
+            }
+        }
+    ]
+}
+-->
+
+* When **id** and **colors** are both omitted: Returns a string array containing all the known rainbow palette IDs.
+* When **id** is supplied and **colors** is omitted: Returns a function that can be used to convert a numeric value to a color.
+* When **id** and **colors** are both supplied creates a new palette instance based on the supplied colors and returns a function that can be used to convert a numeric value to a color.
```

Comment:
`Palette.rainbow.fetchRainbowItem("mySpectrum", ["red", "white", "blue"]):`
Also, shouldn't we mention "steps" here? Something like...
"A third argument may be supplied to indicate the number of desired color samples (aka 'steps')"